hanjaelee/app.js

```diff
@@ -52,20 +53,175 @@ app.post('/signup', (req, res) => {
       profile_image,
       password
     ) VALUES (?, ?, ?, ?);`
-  const rawData = database.query(rawQuery, [name, email, profileImage, password])
 
-  rawData
-    .then((data) => {
-      if (data.affectedRows) {
-        res.status(201).json({ message: 'userCreated' })
-        return
-      }
+  try {
+    const rawData = await database.query(rawQuery, [name, email, profileImage, password])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'userCreated' })
+    }
+    return res.status(202).json({ message: 'userNotCreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 3 게시글 등록하기
+app.post('/post', async (req, res) => {
+  const { title, content, userId } = req.body
+
+  const rawQuery =
+    `
+    INSERT INTO posts (
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);`
+
+  try {
+    const rawData = await database.query(rawQuery, [title, content, userId])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
```

Comment:
데이터 받아오는 부분도 `try { ... }` 구문 안에 넣어도 될 것 같습니다.
이 부분도 마찬가지로 포매팅 한번 수정해서 가독성 높이고 에러 디버깅 코드 추가하면 좋을 것 같습니다.

```suggestion
  try {
  const { title, content, userId } = req.body

  const rawQuery =
    `
    INSERT INTO posts (
      title,
      content,
      user_id
    ) VALUES (?, ?, ?);`

    const rawData = await database.query(rawQuery, [title, content, userId])
    
    if (rawData['affectedRows']) {
      return res.status(201).json({ message: 'postCreated' })
    }
    
    return res.status(202).json({ message: 'postnotcreated' })
  } catch (error) {
    console.error("Failed to Posting", error)
    return res.status(400).json({ message: error.sqlMessage })
  }
```